### PR TITLE
feat!: upgrade to Remix 2

### PR DESCRIPTION
**Pull request by ascorbic**
## Description

Update adapters and edge runtime to Remix 2, and to Functions API v2. Switches to using our custom server during dev in edge, now that there's better support for this. Moves everything to ESM now that's required. 

There is a separate PR to update the template once this is out.

## Related Tickets & Documents

- Related Issue #
- Fixes FRA-48

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes_

For us to review and ship your PR efficiently, please perform the following steps:

- [ ] Open a [bug/issue](https://github.com/netlify/remix-compute/issues/new/choose) before writing your code 🧑‍💻. This
      ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a
      typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 📖. This ensures your code follows our style
      guide and passes our tests.
- [ ] Update or add tests (if any source code was changed or added) 🧪
- [ ] Update or add documentation (if features were changed or added) 📝
- [ ] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: packages/demo-site/app/entry.server.tsx [hunk 1]
```diff
@@ -4,132 +4,77 @@
...
+            pipe(body)
+          }
         },
         onShellError(error: unknown) {
-          reject(error);
+          reject(error)
+        },
+        onAllReady() {
+          // Avoid a bug where responses aren't flushed if there's an outstanding timer.
+          clearTimeout(timer)
```
Comment (ascorbic) [inline packages/demo-site/app/entry.server.tsx:51]: https://github.com/netlify/pod-dev-foundations/issues/604

File: packages/demo-site/remix.config.js [hunk 2]
```diff
@@ -1,18 +1,7 @@
 /** @type {import('@remix-run/dev').AppConfig} */
-module.exports = {
+export default {
   ignoredRouteFiles: ['**/.*'],
-  server: process.env.NETLIFY || process.env.NETLIFY_LOCAL ? './server.ts' : undefined,
```
Comment (ascorbic) [inline packages/demo-site/remix.config.js:4]: We can now use our custom server in dev

**Comment by ascorbic** [comment]
> ### ❌ Deploy Preview for _hydrogen-demo-site_ failed.
> Name 	Link
> 🔨 Latest commit 	[77694d9](https://github.com/netlify/remix-compute/commit/77694d97be31f49e8063037d64c331760a84fc82)
> 🔍 Latest deploy log 	https://app.netlify.com/sites/hydrogen-demo-site/deploys/652f7277b6c7780008aff8cd

This site no longer exists

**Review by pieh — DISMISSED** [review]
Seems good - need to resolve `package-lock.json` conflict and will have to re-approve after that

**Review by pieh — DISMISSED** [review]

**Review by pieh — APPROVED** [review]

File: packages/demo-site/.eslintrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": ["@remix-run/eslint-config", "@remix-run/eslint-config/node"],
+}
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply packages/demo-site/.eslintrc:3]: ESLint v9 is going to remove support for JSON `.eslintrc` files

**Pull request merged**